src/main/java/org/reactome/addlinks/referencecreators/TargetPathogenReferenceCreator.java

```diff
@@ -0,0 +1,219 @@
+
+import org.gk.model.GKInstance;
+import org.gk.model.ReactomeJavaConstants;
+import org.gk.persistence.MySQLAdaptor;
+import org.gk.schema.InvalidAttributeException;
+
+public class TargetPathogenReferenceCreator extends SimpleReferenceCreator<String>
+{
+	private static final String REACTOME_IDENTIFIER_PATTERN_STRING = "R-[A-Z]{3}-.*";
+	private static final Pattern REACTOME_IDENTIFIER_PATTERN = Pattern.compile(REACTOME_IDENTIFIER_PATTERN_STRING);
+
+	// These sets are static so that the first instance of this class can populate them, and all other instances can
+	// just rely on the data that has already been set.
+	private static Map<String, Set<String>> reactomeToTargetPathogen = new HashMap<>();
+	private static Map<String, Set<String>> uniProtToTargetPathogen = new HashMap<>();
+
+	private long personID;
+
+	public TargetPathogenReferenceCreator(MySQLAdaptor adapter, String classToCreate, String classReferring, String referringAttribute, String sourceDB, String targetDB)
+	{
+		super(adapter, classToCreate, classReferring, referringAttribute, sourceDB, targetDB);
+	}
+
+	public TargetPathogenReferenceCreator(MySQLAdaptor adapter, String classToCreate, String classReferring, String referringAttribute, String sourceDB, String targetDB, String refCreatorName)
+	{
+		super(adapter, classToCreate, classReferring, referringAttribute, sourceDB, targetDB, refCreatorName);
+	}
+
+	/**
+	 * Can create references to Target Pathogen on reaction objects that are identifier by a Reactome identifier. We will also probably want to create a reference
+	 * from proteins identified by UniProt identifiers. Might be tricky to combine this functionality into one ReferenceCreator, but we'll see...
+	 */
+	@Override
+	public void createIdentifiers(long personID, Map<String, String> mapping, List<GKInstance> sourceReferences) throws Exception
+	{
+		this.personID = personID;
+		if (reactomeToTargetPathogen.size() == 0 && uniProtToTargetPathogen.size() == 0)
+		{
+			this.logger.info("Identifier sets are empty, populating them now...");
+
+			// the maps are static, shared between instances. Synchronized block should ensure that they cannot be accessed asynchronously.
+			synchronized (this)
+			{
+				for (String targetPathogenIdentifier : mapping.keySet())
+				{
+					String[] mappedIdentifierParts = mapping.get(targetPathogenIdentifier).split("\\|");
+					for (String identifier : mappedIdentifierParts)
+					{
+						// If the string matches the Reactome pattern, it is a Reactome identifier.
+						if (REACTOME_IDENTIFIER_PATTERN.matcher(identifier).matches())
+						{
+							if (!reactomeToTargetPathogen.containsKey(identifier))
+							{
+								reactomeToTargetPathogen.put(identifier, new HashSet<>(Arrays.asList(targetPathogenIdentifier)));
+							}
+							else
+							{
+								Set<String> tmpSet = reactomeToTargetPathogen.get(identifier);
+								tmpSet.add(targetPathogenIdentifier);
+								reactomeToTargetPathogen.put(identifier, tmpSet);
```

Comment:
Could this be written as:
```
reactomeToTargetPathogen.get(identifier).add(targetPathogenIdentifier);
```

Reply:
It might be possible to replace the first two lines with one line, but I'm not sure what the benefit is. I sometimes find such expressions can be harder to debug when NPEs or other issues pop up.

Reply:
I find it easier to understand what is going on without the creation and re-assignment of the temporary set.

For this logic, I don't think a NullPointerException should ever happen because this happens after a check that the identifier does exist and so the `reactomeToTargetPathogen.get(identifier)` should always return a set that you can add to.

Reply:
Personally, I like the explicitness of it. And sure, an NPE probably should never happen, but I've been trying to get into this habit because in other situations, too many chained calls has resulted in having to split everything up to trace exactly where the error was. So now I'm trying to get _out_ of the habit of too many chained function calls.

Reply:
Okay, I'll leave it up to you.  

I agree that multiple chained calls can be difficult to debug.  I was just thinking in this case it would be okay for readability since it is only one chained call and the first method call is guaranteed to return a non-null value.

Reply:
I realized you could do this:
`reactomeToTargetPathogen.computeIfAbsent(identifier, k -> new HashSet()).add(targetPathogenIdentifier);`

The "computeIfAbsent" will guarantee a key that isn't found will create an empty set to be added to and guarantee avoidance of a NPE

Reply:
I still don't like the chained `.add` here. And I'm not sure I like `computeIfAbsent` either. In this context, it seems to simply hide a very simple logical expression. I'm not sure I see the value of it in this situation.

Reply:
Okay, no problem if you prefer the current implementation.  It's just my preference not to reassign to a map if a value already exists.